### PR TITLE
feat: relay burn event

### LINKED ISSUE #1: Relaying burn event to 9c network
- [x] Subscribe wNCG `Burn` event from Ethereum network.
- [ ] Relay the events into Nine Chronicles network through GrpahQL API.

### PR DESCRIPTION
**Pull request by moreal**
It resolves #1, #2

It does
 - re-initialize project with TypeScript + web3.js.
 - `wNCG` solidity definitions. I published on ropsten to test. See [Ropsten network contract](https://ropsten.etherscan.io/address/0xbe865B578cfdd4DCc35Be44B9a90F1b6176c7D61)
 - In bridge app, it subscribes `Burn` event of `wNCG` with 10 blocks confirmations and transfer NCG through `INCGTransfer` interface. But it is not implemented yet and it will be implemented in next pull request.

### CODE REVIEW

File: contracts/contracts/WrappedNCGToken.sol [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+pragma solidity >=0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract WrappedNCGToken is ERC20, Ownable {
+    event Burn(address indexed _sender, bytes32 indexed _to, uint256 amount);
+
+    constructor() public ERC20("Wrapped NCG Token", "wNCG") {}
```
Comment (jckdotim) [inline contracts/contracts/WrappedNCGToken.sol:11]:
```suggestion
    constructor() public ERC20("Wrapped NCG", "wNCG") {}
```

Why don't we remove `Token` like [WBTC](https://github.com/WrappedBTC/bitcoin-token-smart-contracts/blob/master/ethereumV2/contracts/token/WBTC.sol#L11)?

File: contracts/contracts/WrappedNCGToken.sol [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract WrappedNCGToken is ERC20, Ownable {
```
Comment (jckdotim) [inline contracts/contracts/WrappedNCGToken.sol:8]:
```suggestion
contract WrappedNCG is ERC20, Ownable {
```

ditto

File: contracts/migrations/2_deploy_wrap_ncg_token.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const WrappedNCGToken = artifacts.require("WrappedNCGToken");
```
Comment (jckdotim) [inline contracts/migrations/2_deploy_wrap_ncg_token.js:1]:
```suggestion
const WrappedNCG = artifacts.require("WrappedNCG");
```

ditto

File: contracts/migrations/2_deploy_wrap_ncg_token.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+
+module.exports = function(deployer) {
+  deployer.deploy(WrappedNCGToken);
```
Comment (jckdotim) [inline contracts/migrations/2_deploy_wrap_ncg_token.js:4]:
```suggestion
  deployer.deploy(WrappedNCG);
`````

ditto

**Review by jckdotim — COMMENTED** [review]
Overall is good. Here are minor comments.

File: bridge/src/interfaces/ncg-transfer.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export interface INCGTransfer {
+    transfer(address: string, amount: BigInt): void;
+}
```
[the remark this answers is not in the document]
Comment (moreal) [reply bridge/src/interfaces/ncg-transfer.ts:3]: I wanted to separate `transfer` definition with this interface to be easy for test in later.

**Review by jckdotim — APPROVED** [review]
LGTM

**Review by longfin — APPROVED** [review]

**Pull request merged**